manage_breast_screening/record_a_mammogram/forms.py

```diff
@@ -43,42 +43,33 @@ def save(self):
+        ("failed_identity_check", "Failed identity check"),
+        ("language_difficulties", "Language difficulties"),
+        ("physical_health_issue", "Physical health issue"),
+        ("mental_health_issue", "Mental health issue"),
+        ("last_mammogram_within_6_months", "Last mammogram within 6 months"),
+        ("breast_implant_risks", "Breast implant risks"),
+        ("pain_during_screening", "Pain during screening"),
+        ("technical_issues", "Technical issues"),
+        ("participant_withdrew_consent", "Participant withdrew consent"),
+        ("other", "Other"),
+    )
+
     def __init__(self, *args, **kwargs):
-        self.instance = kwargs.pop('instance', None)
+        if 'instance' not in kwargs:
+            raise ValueError("AppointmentCannotGoAheadForm requires an instance")
+        self.instance = kwargs.pop('instance')
         super().__init__(*args, **kwargs)
 
-    participant_did_not_attend = forms.BooleanField(required=False, label="Participant did not attend")
-    participant_did_not_attend_details = forms.CharField(required=False)
-
-    failed_identity_check = forms.BooleanField(required=False, label="Failed identity check")
-    failed_identity_check_details = forms.CharField(required=False)
-
-    language_difficulties = forms.BooleanField(required=False, label="Language difficulties")
-    language_difficulties_details = forms.CharField(required=False)
-
-    physical_health_issue = forms.BooleanField(required=False, label="Physical health issue")
-    physical_health_issue_details = forms.CharField(required=False)
-
-    mental_health_issue = forms.BooleanField(required=False, label="Mental health issue")
-    mental_health_issue_details = forms.CharField(required=False)
-
-    last_mammogram_within_6_months = forms.BooleanField(required=False, label="Last mammogram within 6 months")
-    last_mammogram_within_6_months_details = forms.CharField(required=False)
+        # Dynamically add detail fields for each choice
+        for field_name, _ in self.STOPPED_REASON_CHOICES:
+            self.fields[f"{field_name}_details"] = forms.CharField(required=False)
 
-    breast_implant_risks = forms.BooleanField(required=False, label="Breast implant risks")
-    breast_implant_risks_details = forms.CharField(required=False)
-
-    pain_during_screening = forms.BooleanField(required=False, label="Pain during screening")
-    pain_during_screening_details = forms.CharField(required=False)
-
-    technical_issues = forms.BooleanField(required=False, label="Technical issues")
-    technical_issues_details = forms.CharField(required=False)
-
-    participant_withdrew_consent = forms.BooleanField(required=False, label="Participant withdrew consent")
-    participant_withdrew_consent_details = forms.CharField(required=False)
-
-    other = forms.BooleanField(required=False, label="Other")
-    other_details = forms.CharField(required=False)
+    stopped_reasons = forms.MultipleChoiceField(
```

Comment:
this refactoring looks a lot cleaner than the previous version 👍🏻 

Thinking out loud, if we end up reusing this pattern, we could potentially go one step further and factor out the reasons checkbox + details elements into a custom field, whose value would be the dictionary currently built in the `save()` method. But the dynamically generated field approach seems like it will work.